### PR TITLE
feat: check commit messages to see if a new version needs to be published

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?
Checks commit messages to see if we need to publish a new version or not. For example
  - if only a `chore` commit is found, then it will exit early.
  - if both a `chore` and `feat` commit is found, then it will continue with the release

TODO:
- [x] fix broken unit tests

### What issues does this PR fix or reference?
@W-8775832@
@W-8820671@

### CODE REVIEW

**Review by peternhale — APPROVED** [review]

File: src/repository.ts
Comment (amphro) [inline src/repository.ts:206]: Can we put in the comment that this is a workaround with a link to the issue on standard-version. I would like to remove this down the road if they update that library.

File: src/repository.ts
Comment (amphro) [inline src/repository.ts:447]: So here is a question. If we force update the version in `package.json` without a `fit` or `feat` commit message, do we still want to release it? For example, maybe in `salesforcedx` we have a commit message on Thursday as `chore: release 50.17.0` where we hardcode the version in that commit. In that case we want to release. It feels weird to have the commit be `feat: release 50.17.0`.

Thoughts?

**Review by amphro — CHANGES_REQUESTED** [review]

File: src/repository.ts
Comment (amphro) [reply src/repository.ts:447]: Maybe to stay consistent we do use `feat` to kick off the release... just pointing it out.

**Comment by amphro** [comment]
Can we also add some information to a readme somewhere if it isn't already? Seems like it should be doc'ed

File: src/repository.ts
Comment (mdonnalley) [reply src/repository.ts:447]: That's a good point. I'll update it to ignore this check if the version bump has been hardcoded in the package.json

File: README.md
Comment (amphro) [inline README.md:15]: Sweet. Thanks.

File: src/repository.ts
Comment (amphro) [inline src/repository.ts:26]: Why do these need to be brought in as require? Can't you still do `import * as conventionalCommitsParser from 'conventional-commits-parser'`?

File: src/repository.ts
Comment (amphro) [inline src/repository.ts:209]: Thank you.

**Review by amphro — APPROVED** [review]

File: src/repository.ts
Comment (shetzel) [inline src/repository.ts:352]: nit: `get()` --> `getString()`

**Review by shetzel — APPROVED** [review]
just some nits, but approved

**Comment by amphro** [comment]
*Testing notes:*

If I have *no type* or a *random invalid type*, it will still publish. You could argue that is a problem with the repo not validating commit messages but it could also be bypassed by the command line or UI. In that case, maybe it should be published. If the answer is on, Instead of `skippableCommitTypes` we should have `commitTypes: ['feat', 'feat!', 'bug', 'bug!', 'perf', 'refactor']`.  I'm fine keeping it skippable too. Either way, it is a list we need to maintain.

I published ~15 package versions with a variety of commit messages in between each release testing patch, minor, and major. Other than the issue above it works great. 

I'll wait for @mdonnalley feedback on the issue before merging.

**Comment by mdonnalley** [comment]
@amphro I think it's fine to allow it to publish in those scenarios. If someone has gone through the trouble of bypassing the commit linting, then they are probably (hopefully) doing it purposely. That being said, we can always address that scenario later if it becomes an issue

**Comment by amphro** [comment]
Sounds good to me. I just wanted to make sure we had the conversation.

**Pull request merged**
